src/couch_util.erl

```diff
@@ -584,3 +585,14 @@ with_proc(M, F, A, Timeout) ->
         erlang:demonitor(Ref, [flush]),
         {error, timeout}
     end.
+
+
+take_kv_values(List, Keys) ->
+    lists:foldr(fun(Key, {Acc, KVList}) ->
```

Comment:
derp, answered my own question: A body-recursive list function and a tail-recursive function that calls lists:reverse/1 at the end will use the same amount of memory.